lib/Doctrine/Migrations/Generator/SqlGenerator.php

```diff
@@ -30,10 +31,14 @@ class SqlGenerator
     /** @var AbstractPlatform */
     private $platform;
 
+    /** @var SqlFormatter */
+    private $sqlFormatter;
+
     public function __construct(Configuration $configuration, AbstractPlatform $platform)
     {
         $this->configuration = $configuration;
         $this->platform      = $platform;
+        $this->sqlFormatter  = new SqlFormatter(new NullHighlighter());
```

Comment:
creating the sql formatter should be done on `generate` when `$formatted=true`, having it in the constructor makes is a hard dependency

Reply:
should we move the `if (! class_exists(SqlFormatter::class)) {` checks inside this class?

Reply:
The exception talks about a CLI option, so I don't think we should.